### DISCUSSION TITLE
Standalone usage

### DISCUSSION THREAD

**Discussion by m1h43l** [infinispan/protostream#258] [Q&A]
Is it possible to use protostream standalone without an Infinispan cache? Just for serialization and deserialization from/to a byte array or ByteBuffer?

**Comment by tristantarrant** [comment]
Protostream does not depend on Infinispan classes/caches, so there should be no problem. Let us know if it fits your use case

**Comment by m1h43l** [comment]
Thanx. I am trying to get a little prototype running.

ProtobufUtils seems like the main class to use. Do I have to manually create the proto files for my Java classes and register them at the SerializationContext? And do I have to code a Marshaller and register that too?

Currently I got the following proto:

```
syntax = "proto2";

package werewolf.server.common;

message User {
	optional int64 userId = 1;
	optional string name = 2;
	optional string email = 3;
}
```

Code:
``` java
User user = new User();
user.setUserId(1l);
user.setEmail("mihael@rpgnextgen.com");
user.setLoginToken(null);
user.setName("Mihael");

SerializationContext context = ProtobufUtil.newSerializationContext();
context.registerProtoFiles(FileDescriptorSource.fromResources("protos/User.proto"));
Assertions.assertTrue(context.canMarshall(user));
```

Any help is very appreciated.
